### PR TITLE
Named argument in cpp call to Sys.setenv()

### LINKED ISSUE #94: Error in anytime, with useR = TRUE, Windows only
This only appeared in a recent version. `UseR = TRUE` is broken on Windows.

### Win

``` r
anytime::anytime("2001-01-01", useR = TRUE)
#> Error in anytime_cpp(x, tz = tz, asUTC = asUTC, useR = useR, oldHeuristic = oldHeuristic): Evaluation error: all arguments must be named.
```

<sup>Created on 2019-06-25 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.5.3 (2019-03-11)
#>  os       Windows Server x64          
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       Europe/Berlin               
#>  date     2019-06-25                  
#> 
#> - Packages --------------------------------------------------------------
#>  package      * version date       lib source        
#>  anytime        0.3.4   2019-06-18 [1] CRAN (R 3.5.3)
#>  assertthat     0.2.1   2019-03-21 [1] CRAN (R 3.5.3)
#>  backports      1.1.4   2019-04-10 [1] CRAN (R 3.5.3)
#>  callr          3.2.0   2019-03-15 [1] CRAN (R 3.5.3)
#>  cli            1.1.0   2019-03-19 [1] CRAN (R 3.5.3)
#>  crayon         1.3.4   2017-09-16 [1] CRAN (R 3.5.3)
#>  desc           1.2.0   2018-05-01 [1] CRAN (R 3.5.3)
#>  devtools       2.0.2   2019-04-08 [1] CRAN (R 3.5.3)
#>  digest         0.6.19  2019-05-20 [1] CRAN (R 3.5.3)
#>  evaluate       0.14    2019-05-28 [1] CRAN (R 3.5.3)
#>  fs             1.3.1   2019-05-06 [1] CRAN (R 3.5.3)
#>  glue           1.3.1   2019-03-12 [1] CRAN (R 3.5.3)
#>  highr          0.8     2019-03-20 [1] CRAN (R 3.5.3)
#>  htmltools      0.3.6   2017-04-28 [1] CRAN (R 3.5.3)
#>  knitr          1.23    2019-05-18 [1] CRAN (R 3.5.3)
#>  magrittr       1.5     2014-11-22 [1] CRAN (R 3.5.3)
#>  memoise        1.1.0   2017-04-21 [1] CRAN (R 3.5.3)
#>  pkgbuild       1.0.3   2019-03-20 [1] CRAN (R 3.5.3)
#>  pkgload        1.0.2   2018-10-29 [1] CRAN (R 3.5.3)
#>  prettyunits    1.0.2   2015-07-13 [1] CRAN (R 3.5.3)
#>  processx       3.3.1   2019-05-08 [1] CRAN (R 3.5.3)
#>  ps             1.3.0   2018-12-21 [1] CRAN (R 3.5.3)
#>  R6             2.4.0   2019-02-14 [1] CRAN (R 3.5.3)
#>  RApiDatetime   0.0.4   2018-10-21 [1] CRAN (R 3.5.2)
#>  Rcpp           1.0.1   2019-03-17 [1] CRAN (R 3.5.3)
#>  remotes        2.1.0   2019-06-24 [1] CRAN (R 3.5.3)
#>  rlang          0.3.4   2019-04-07 [1] CRAN (R 3.5.3)
#>  rmarkdown      1.13    2019-05-22 [1] CRAN (R 3.5.3)
#>  rprojroot      1.3-2   2018-01-03 [1] CRAN (R 3.5.3)
#>  sessioninfo    1.1.1   2018-11-05 [1] CRAN (R 3.5.3)
#>  stringi        1.4.3   2019-03-12 [1] CRAN (R 3.5.3)
#>  stringr        1.4.0   2019-02-10 [1] CRAN (R 3.5.3)
#>  testthat       2.1.1   2019-04-23 [1] CRAN (R 3.5.3)
#>  usethis        1.5.0   2019-04-07 [1] CRAN (R 3.5.3)
#>  withr          2.1.2   2018-03-15 [1] CRAN (R 3.5.3)
#>  xfun           0.8     2019-06-25 [1] CRAN (R 3.5.3)
#>  yaml           2.2.0   2018-07-25 [1] CRAN (R 3.5.2)
#> 
#> [1] D:/Users/christoph/Documents/R/win-library/3.5
#> [2] C:/Program Files/R/R-3.5.3/library
```

</details>

### OS-X (and probably Linux)

``` r
anytime::anytime("2001-01-01", useR = TRUE)
#> [1] "2001-01-01 CET"
```

<sup>Created on 2019-06-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.0 (2019-04-26)
#>  os       macOS Mojave 10.14.5        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       Europe/Zurich               
#>  date     2019-06-25                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package      * version date       lib source        
#>  anytime        0.3.4   2019-06-18 [1] CRAN (R 3.6.0)
#>  assertthat     0.2.1   2019-03-21 [1] CRAN (R 3.6.0)
#>  backports      1.1.4   2019-04-10 [1] CRAN (R 3.6.0)
#>  callr          3.2.0   2019-03-15 [1] CRAN (R 3.6.0)
#>  cli            1.1.0   2019-03-19 [1] CRAN (R 3.6.0)
#>  crayon         1.3.4   2017-09-16 [1] CRAN (R 3.6.0)
#>  desc           1.2.0   2018-05-01 [1] CRAN (R 3.6.0)
#>  devtools       2.0.2   2019-04-08 [1] CRAN (R 3.6.0)
#>  digest         0.6.19  2019-05-20 [1] CRAN (R 3.6.0)
#>  evaluate       0.13    2019-02-12 [1] CRAN (R 3.6.0)
#>  fs             1.3.0   2019-05-02 [1] CRAN (R 3.6.0)
#>  glue           1.3.1   2019-03-12 [1] CRAN (R 3.6.0)
#>  highr          0.8     2019-03-20 [1] CRAN (R 3.6.0)
#>  htmltools      0.3.6   2017-04-28 [1] CRAN (R 3.6.0)
#>  knitr          1.22    2019-03-08 [1] CRAN (R 3.6.0)
#>  magrittr       1.5     2014-11-22 [1] CRAN (R 3.6.0)
#>  memoise        1.1.0   2017-04-21 [1] CRAN (R 3.6.0)
#>  pkgbuild       1.0.3   2019-03-20 [1] CRAN (R 3.6.0)
#>  pkgload        1.0.2   2018-10-29 [1] CRAN (R 3.6.0)
#>  prettyunits    1.0.2   2015-07-13 [1] CRAN (R 3.6.0)
#>  processx       3.3.0   2019-03-10 [1] CRAN (R 3.6.0)
#>  ps             1.3.0   2018-12-21 [1] CRAN (R 3.6.0)
#>  R6             2.4.0   2019-02-14 [1] CRAN (R 3.6.0)
#>  RApiDatetime   0.0.4   2018-10-21 [1] CRAN (R 3.6.0)
#>  Rcpp           1.0.1   2019-03-17 [1] CRAN (R 3.6.0)
#>  remotes        2.0.4   2019-04-10 [1] CRAN (R 3.6.0)
#>  rlang          0.3.4   2019-04-07 [1] CRAN (R 3.6.0)
#>  rmarkdown      1.12    2019-03-14 [1] CRAN (R 3.6.0)
#>  rprojroot      1.3-2   2018-01-03 [1] CRAN (R 3.6.0)
#>  sessioninfo    1.1.1   2018-11-05 [1] CRAN (R 3.6.0)
#>  stringi        1.4.3   2019-03-12 [1] CRAN (R 3.6.0)
#>  stringr        1.4.0   2019-02-10 [1] CRAN (R 3.6.0)
#>  testthat       2.1.1   2019-04-23 [1] CRAN (R 3.6.0)
#>  usethis        1.5.0   2019-04-07 [1] CRAN (R 3.6.0)
#>  withr          2.1.2   2018-03-15 [1] CRAN (R 3.6.0)
#>  xfun           0.6     2019-04-02 [1] CRAN (R 3.6.0)
#>  yaml           2.2.0   2018-07-25 [1] CRAN (R 3.6.0)
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

### PR DESCRIPTION
**Pull request by christophsax**
- fixes #94
- also mentions all arguments in `anytime_cpp`

### CODE REVIEW

File: R/anytime.R [hunk 1]
```diff
@@ -175,7 +175,7 @@ anytime <- function(x, tz=getTZ(), asUTC=FALSE, useR=FALSE,
         x <- as.character(x)
     }
 
-    anytime_cpp(x, tz=tz, asUTC=asUTC, useR=useR, oldHeuristic=oldHeuristic)
+    anytime_cpp(x, tz=tz, asUTC=asUTC, asDate=FALSE, useR=useR, oldHeuristic=oldHeuristic)
```
Comment (eddelbuettel) [inline R/anytime.R:178]: That's actually not needed, but does not hurt.

Comment (eddelbuettel) [reply R/anytime.R:178]: And once we have it. we need the same in the two other calls to `anytime_cpp`.  Or in none.

And the other two may of course need other values for date and utc. :)

File: src/anytime.cpp [hunk 2]
```diff
@@ -342,7 +342,7 @@ double r_stringToTime(const std::string s, const std::string tz,
     }
 #ifdef _WIN32
     Rcpp::Function f("Sys.setenv");
-    f("TZ", (oldtz == nullptr ? "" : oldtz), 1);
+    f(Rcpp::Named("TZ") = (oldtz == nullptr ? "" : oldtz));
```
Comment (eddelbuettel) [inline src/anytime.cpp:345]: Thanks so what I suggested in #94 carried the day.

Thanks a bunch for finding it, and the PR.

**Review by eddelbuettel — CHANGES_REQUESTED** [review]

File: R/anytime.R (hunk 1 above)
Comment (christophsax) [reply R/anytime.R:178]: But they had the `asDate=` argument already? 

Line 190:
```r
d <- anytime_cpp(x=x, tz=tz, asUTC=asUTC, asDate=TRUE, useR=useR, oldHeuristic=TRUE)
```

Line 214:
```r
 d <- anytime_cpp(x=x, tz=tz, asUTC=TRUE, asDate=TRUE, useR=useR, oldHeuristic=TRUE)
```

Comment (eddelbuettel) [reply R/anytime.R:178]: Oh in which case we're done indeed.  My bad -- we're good.

**Review by eddelbuettel — APPROVED** [review]

**Comment by christophsax** [comment]
Thanks for the light-speed support!

Any plans that this goes to CRAN soon? My win ERRORS in tsbox are coming from this one (the Solaris one is unrelated): https://cran.r-project.org/web/checks/check_results_tsbox.html

**Comment by eddelbuettel** [comment]
Yes.  Was thinking within days as [Solaris blows up on CRAN for me](https://cloud.r-project.org/web/checks/check_results_anytime.html) (which I "addressed" by suppressing tests).

I was wondering if I could lean on you for one more quick test.  I also suppressed a lot of Windows tests, maybe too many.  With `tinytest`, you can very easily run individual test files via `Rscript -e 'tinytest::run_test_file("nameofile.R")'`.  Could you open the files in `inst/tinytest/` that `grep` for `isWindows` and see if some of the tests could in fact be reactivated?  I could test it too, but only via `rhub` and takes a few cycles...

**Comment by christophsax** [comment]
Yes, will do. But only have time in in a few hours.

**Comment by eddelbuettel** [comment]
Thanks so much -- that works for me. I'll pick it up in 12 hours after work.

**Pull request merged**